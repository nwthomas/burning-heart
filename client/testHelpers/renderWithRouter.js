import React from "react";
import { NativeRouter as Router } from "react-router-native";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};

export default renderWithRouter;
