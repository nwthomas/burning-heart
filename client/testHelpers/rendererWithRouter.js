import React from "react";
import { NativeRouter as Router } from "react-router-native";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(<Router history={history}>{ui}</Router>),
    history
  };
};

export default rendererWithRouter;
