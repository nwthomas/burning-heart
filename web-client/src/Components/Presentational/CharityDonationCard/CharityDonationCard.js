import React from "react";
import { connect } from "react-redux";

const CharityDonationCard = ({ donation }) => {
  return (
    <div className="donation__card">
      <p className="donation__amount">{`$${donation.amount / 100}`}</p>
    </div>
  );
};

export default connect(
  null,
  {}
)(CharityDonationCard);
