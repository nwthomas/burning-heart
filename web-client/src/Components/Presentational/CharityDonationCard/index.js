export { default as CharityDonationCard } from "./CharityDonationCard";
