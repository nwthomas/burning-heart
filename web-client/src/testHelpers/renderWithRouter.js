import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default renderWithRouter;
