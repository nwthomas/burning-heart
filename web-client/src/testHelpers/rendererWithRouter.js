import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default rendererWithRouter;
