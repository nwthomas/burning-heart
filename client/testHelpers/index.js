import renderWithRouter from "./renderWithRouter";
import rendererWithRouter from "./rendererWithRouter";

export { renderWithRouter, rendererWithRouter };
