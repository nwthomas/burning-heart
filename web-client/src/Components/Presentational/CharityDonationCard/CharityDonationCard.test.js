import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../../testHelpers";

import CharityDonationCard from "./CharityDonationCard";

describe("<CharityDonationCard />", () => {
  it("should render the CharityDonationCard to the screen without crashing", () => {
    const donation = { amount: 1000 };
    const helpers = renderWithRouter(
      <CharityDonationCard donation={donation} />
    );
  });

  it("should match the snapshot of CharityDonationCard", () => {
    const donation = { amount: 1000 };
    const tree = rendererWithRouter(
      <CharityDonationCard donation={donation} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
