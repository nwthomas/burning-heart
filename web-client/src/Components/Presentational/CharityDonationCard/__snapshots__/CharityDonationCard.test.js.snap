// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharityDonationCard /> should match the snapshot of CharityDonationCard 1`] = `
<div
  className="donation__card"
>
  <p
    className="donation__amount"
  >
    $10
  </p>
</div>
`;
